### PR TITLE
elf: move local symbol count to the correct place

### LINKED ISSUE #69: When updating faerie I get a linker error on linux
```
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64" "-L" "sysroot/lib/rustlib/x86_64-unknown-linux-gnu/lib" "/home/bjorn/Documenten/rustc_codegen_cranelift/build_sysroot/target/x86_64-unknown-linux-gnu/debug/deps/std.dummy_name.rcgu.o" "-o" "/home/bjorn/Documenten/rustc_codegen_cranelift/build_sysroot/target/x86_64-unknown-linux-gnu/debug/deps/libstd.so" "/home/bjorn/Documenten/rustc_codegen_cranelift/build_sysroot/target/x86_64-unknown-linux-gnu/debug/deps/std.allocator_shim.rcgu.o" "-Wl,-zrelro" "-Wl,-znow" "-nodefaultlibs" "-L" "/home/bjorn/Documenten/rustc_codegen_cranelift/build_sysroot/target/x86_64-unknown-linux-gnu/debug/deps" "-L" "/home/bjorn/Documenten/rustc_codegen_cranelift/build_sysroot/target/debug/deps" "-L" "sysroot/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-ldl" "-lrt" "-lpthread" "-Wl,-Bstatic" "-Wl,--whole-archive" "/tmp/rustc5pSiro/libpanic_abort-ebe895f026b5ad95.rlib" "-Wl,--no-whole-archive" "-Wl,--whole-archive" "/tmp/rustc5pSiro/libunwind-9bf4ad135fcef4eb.rlib" "-Wl,--no-whole-archive" "-Wl,--whole-archive" "/tmp/rustc5pSiro/librustc_demangle-47343bf5c05ad1ab.rlib" "-Wl,--no-whole-archive" "-Wl,--whole-archive" "/tmp/rustc5pSiro/liblibc-46063ca8474f398d.rlib" "-Wl,--no-whole-archive" "-Wl,--whole-archive" "/tmp/rustc5pSiro/librustc_std_workspace_core-96d00f6d22747389.rlib" "-Wl,--no-whole-archive" "-Wl,--whole-archive" "/tmp/rustc5pSiro/liballoc-8684167332e0bd28.rlib" "-Wl,--no-whole-archive" "-Wl,--whole-archive" "/tmp/rustc5pSiro/libcore-1958aff3d7d300b8.rlib" "-Wl,--no-whole-archive" "/tmp/rustc5pSiro/libcompiler_builtins-2fce88ecd3031979.rlib" "-Wl,-Bdynamic" "-lgcc_s" "-lc" "-lm" "-lrt" "-lpthread" "-lutil" "-lutil" "-shared"
  = note: /tmp/rustc5pSiro/libpanic_abort-ebe895f026b5ad95.rlib: member /tmp/rustc5pSiro/libpanic_abort-ebe895f026b5ad95.rlib(dummy_namercgu) in archive is not an object
          collect2: error: ld returned 1 exit status
```

Repro at https://github.com/bjorn3/rustc_codegen_cranelift/tree/link_error

### PR DESCRIPTION
**Pull request by pchickey**
fixes #69. Bug is my fault, @m4b and @philipc figured out what I messed up.

### CODE REVIEW

File: src/elf.rs [hunk 1]
```diff
@@ -470,9 +470,6 @@ impl<'a> Elf<'a> {
     }
     pub fn add_definition(&mut self, name: &str, data: &'a [u8], decl: &artifact::DefinedDecl) {
         // sh_info requires nsections + nlocals to add as delimiter; see the associated FunFact
```
Comment (m4b) [inline src/elf.rs]: floating comment now

**Comment by m4b** [comment]
it's not your fault; or if it is, don't feel bad _at all_; `sh_info` being **crucially meaningful** in a single section, at just one point, in an otherwise unused field in the object file is horrible (I broke faerie last time with it and took forever to figure out) and linkers are terrible for:

1. supporting it as the way to determine where globals are (why not use, like, uh `ST_BIND` for that?)
2. not printing error messages when its wrong and just segfaulting instead

and thanks for fixing this ! :)

**Comment by bjorn3** [comment]
Thanks @m4b and @philipc for figuring out the problem and @pchickey for fixing it. I will try this out soon.

**Comment by m4b** [comment]
Thank you for the bug report and doing all that great work on rust cranelift backend :D

**Comment by bjorn3** [comment]
This does indeed fix it.

**Comment by pchickey** [comment]
published on crates.io as 0.9.1

**Comment by bjorn3** [comment]
Thanks @pchickey

**Pull request merged**